tests/publiccloud/prepare_tools.pm

```diff
@@ -83,7 +83,6 @@ sub run {
     my $azure_error = '/tmp/azure_error';
     record_info('Azure', script_output('az -v 2>' . $azure_error));
     assert_script_run('cat ' . $azure_error);
-    assert_script_run('test ! -s ' . $azure_error);
```

Comment:
The reason to remove this is basically cause the latest Azure version prints some information output to stderr... Not sure how to workaround it.

Reply:
@cfconrad  WDYT ?

Reply:
Btw, this is the text returned in stderr

```
localhost:~ # cat /tmp/azure_error 

Please let us know how we are doing: https://aka.ms/azureclihats
and let us know if you're interested in trying out our newest features: https://aka.ms/CLIUXstudy
```

Reply:
I can't remove that output.. There is an option introduced in version 2.3.0 `only_show_errors=True ` but we are using az version 2.21...

Reply:
than I would go for 
```perl
if (script_run('test -s ' . $azure_error) ) {

die ("Unexpected error in azure-cli") unless validate_script_output('cat $azure_error', m/Please let us know how we are doing .* and let us know if you're interested in trying out our newest features .*/ );

}
```

Reply:
obviously you will need to work on regex it is just a snippet which I type without actually checking if it working ...

Reply:
> I can't remove that output.. There is an option introduced in version 2.3.0 `only_show_errors=True ` but we are using az version 2.21...

looks like 2.3.0 is not released yet anywhere https://github.com/Azure/azure-cli/releases we using latest available release

Reply:
Misunderstanding. 2.3.0 is lower than 2.21 :) 

We have both tried using the config `az config set core.only_show_errors=true` but it still returns that warning, so I implemented the if-condition and die as suggested.